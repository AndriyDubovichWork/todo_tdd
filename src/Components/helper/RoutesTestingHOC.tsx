import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

type RoutesTestingHOCType = {
  children: JSX.Element;
};
const RoutesTestingHOC = ({ children }: RoutesTestingHOCType) => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path='*' element={children} />
      </Routes>
    </BrowserRouter>
  );
};

export default RoutesTestingHOC;
