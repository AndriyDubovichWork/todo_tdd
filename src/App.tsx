import React, { useEffect, useState } from 'react';
import './App.css';
import ToDo, { TaskType } from './Components/ToDo/ToDo';
import IteractButtons from './Components/IteractButtons/IteractButtons';
import { addToList, DeleteChecked, Delete } from './Components/helper/CruD';
import { Route, Router, Routes } from 'react-router-dom';
import FullScreenSticker from './Components/FullScreenSticker/FullScreenSticker';

function App() {
  const [addTask, setAddTask] = useState({
    text: '',
    id: 404,
    isCompleted: false,
  });

  const [tasks, setTasks]: (TaskType[] | any)[] = useState([]);

  const [isAllSelected, setIsAllSelected]: (boolean | any)[] = useState(false);

  const [isAnySelected, setIsAnySelected]: (boolean | any)[] = useState(false);

  useEffect(() => {
    setIsAllSelected(true);
    setIsAnySelected(false);
    tasks.map((task: TaskType) => {
      if (!task.isCompleted) {
        setIsAllSelected(false);
      } else {
        setIsAnySelected(true);
      }
    });
  }, [tasks, isAnySelected]);

  return (
    <Routes>
      <Route
        path='/todo_tdd'
        element={
          <>
            <IteractButtons
              addToList={addToList}
              addTask={addTask}
              setAddTask={setAddTask}
              setTasks={setTasks}
              DeleteChecked={DeleteChecked}
              tasks={tasks}
              isAllSelected={isAllSelected}
              setIsAllSelected={setIsAllSelected}
              isAnySelected={isAnySelected}
            />
            <ToDo tasks={tasks} setTasks={setTasks} Delete={Delete} />
          </>
        }
      ></Route>
      <Route
        path='/todo_tdd/:toDoId'
        element={
          <>
            <FullScreenSticker tasks={tasks} setTasks={setTasks} />
          </>
        }
      />
    </Routes>
  );
}

export default App;
