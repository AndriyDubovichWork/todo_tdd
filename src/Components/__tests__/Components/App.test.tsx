import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import App from '../../../App';
import RoutesTestingHOC from './../../helper/RoutesTestingHOC';
afterEach(() => cleanup());

test('App have to match snapshot', () => {
  const tree = renderer
    .create(
      <RoutesTestingHOC>
        <App />
      </RoutesTestingHOC>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
