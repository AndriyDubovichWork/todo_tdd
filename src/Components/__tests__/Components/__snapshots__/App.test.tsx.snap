// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App have to match snapshot 1`] = `null`;
